plasma_framework/contracts/poc/fast_exits/Quasar.sol

```diff
@@ -0,0 +1,334 @@
+        uint256 finalizationTimestamp;
+        bool isValid;
+    }
+
+    mapping (address => uint256) public tokenUsableCapacity;
+    mapping (uint256 => Ticket) public ticketData;
+    mapping (uint256 => Claim) private claimData;
+
+    event QuasarTotalEthCapacityUpdated(uint256 balance);
+    
+    modifier onlyQuasarMaintainer() {
+        require(msg.sender == quasarMaintainer, "Only the Quasar Maintainer can invoke this method");
+        _;
+    }
+
+    /**
+     * @dev Constructor, takes params to set up quasar contract
+     * @param plasmaFrameworkContract Plasma Framework contract address
+     * @param _quasarOwner Receiver address on Plasma
+     * @param _safePlasmaBlockNum Safe Blocknum limit 
+     * @param _waitingPeriod Waiting period from submission to processing claim
+     * @param _bondValue bond to obtain tickets
+    */
+    constructor (address plasmaFrameworkContract, address spendingConditionRegistryContract, address _quasarOwner, uint256 _safePlasmaBlockNum, uint256 _waitingPeriod, uint256 _bondValue) public {
+        plasmaFramework = PlasmaFramework(plasmaFrameworkContract);
+        spendingConditionRegistry = SpendingConditionRegistry(spendingConditionRegistryContract);
+        quasarOwner = _quasarOwner;
+        quasarMaintainer = msg.sender;
+        safePlasmaBlockNum = _safePlasmaBlockNum;
+        waitingPeriod = _waitingPeriod;
+        bondValue = _bondValue;
+        bondReserve = 0;
+    }
+
+    ////////////////////////////////////////////	
+    // Maintenance methods	
+    ////////////////////////////////////////////
+    /**
+     * @dev Update the safe blocknum limit
+     * @param newSafePlasmaBlockNum new blocknum limit, has to be higher than previous blocknum limit
+    */
+    function updateSafeBlockLimit (uint256 newSafePlasmaBlockNum) public onlyQuasarMaintainer() {
+        require(newSafePlasmaBlockNum > safePlasmaBlockNum, "New limit should be higher than older limit");
+        safePlasmaBlockNum = newSafePlasmaBlockNum;
+    }
+
+    /**
+     * @dev Flush an expired ticket to free up reserved space
+     * @notice Only an unclaimed ticket can be flushed, bond amount is added to bondReserve
+     * @param utxoPos pos of the output, which is the ticket identifier
+    */
+    function flushExpiredTicket(uint256 utxoPos) public {
+        uint256 expiryTimestamp = ticketData[utxoPos].validityTimestamp;
+        require(!ticketData[utxoPos].isClaimed, "The UTXO has already been claimed");
+        require(block.timestamp > expiryTimestamp && expiryTimestamp != 0, "Ticket still valid or doesn't exist");
+
+        uint256 tokenAmount = ticketData[utxoPos].reservedAmount;
+        ticketData[utxoPos].reservedAmount = 0;
+        ticketData[utxoPos].validityTimestamp = 0;
+        tokenUsableCapacity[ticketData[utxoPos].token] = tokenUsableCapacity[ticketData[utxoPos].token].add(tokenAmount);
+        bondReserve = bondReserve.add(ticketData[utxoPos].bondValue); 
```

Comment:
```suggestion
        tokenUsableCapacity[ticketData[utxoPos].token] = tokenUsableCapacity[ticketData[utxoPos].token].add(tokenAmount);
        bondReserve = bondReserve.add(ticketData[utxoPos].bondValue); 
```
this is basically returning the reserved amount back to the Quasar and claim the bond?

Reply:
Yes! It's just adding to the amount of bonds from expired tickets, that the quasar owner can withdraw.(failed bonds)
I'll redirect you to this for the flow https://github.com/omgnetwork/plasma-contracts/pull/723#issuecomment-737103755